view/src/extras/dealStepper.js

```diff
@@ -0,0 +1,135 @@
+import Typography from "@material-ui/core/Typography";
+import Surplus from "../components/surplus";
+import FoodBanks from "../components/foodbanks";
+import DealForm from "../extras/dealForm";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+  },
+  button: {
+    marginRight: theme.spacing(1),
+  },
+  instructions: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+/** Defines the steps in the Stepper (each is a form) */
+function getSteps() {
+  return [
+    "Select a Surplus Object",
+    "Match with a Food Bank",
+    "Fill out Deal Logistics",
+  ];
+}
+
+export default function HorizontalLinearStepper(props) {
+  const classes = useStyles();
+  // Stepper states
+  const [activeStep, setActiveStep] = React.useState(0);
+
+  const steps = getSteps();
+
+  /** Gets the content to display at each step */
+  const getStepContent = (step) => {
+    switch (step) {
+      case 0:
+        return <Surplus inSurplus={true} alert={props.alert} />;
+      case 1:
+        return <FoodBanks inSurplus={true} alert={props.alert} />;
+      case 2:
+        return <DealForm alert={props.alert} />;
+      default:
+        return "Unknown step";
+    }
+  };
+
+  /** Sets the last step (so formatting can change on last step) */
+  const isLastStep = (step) => {
+    return step === 2;
+  };
+
+  /** Moves to the next step (page) in the Stepper */
+  const handleNext = () => {
+    // Confirm that selected (needs to be connected to child)
+    if (false) {
+      props.alert("warning", "Please select a Surplus Object to continue.");
+      return;
+    }
+
+    setActiveStep((prevActiveStep) => prevActiveStep + 1);
+  };
+
+  /** Moves to the previous step (page) in the Stepper */
+  const handleBack = () => {
+    setActiveStep((prevActiveStep) => prevActiveStep - 1);
+  };
+
+  /** Takes the user back to the first step */
+  const handleReset = () => {
+    setActiveStep(0);
+  };
+
+  return (
+    <div className={classes.root}>
+      <Stepper activeStep={activeStep}>
+        {steps.map((label, index) => {
+          const stepProps = {};
+          const labelProps = {};
+          if (isStepSkipped(index)) {
+            stepProps.completed = false;
+          }
+          return (
+            <Step key={label} {...stepProps}>
+              <StepLabel {...labelProps}>{label}</StepLabel>
+            </Step>
+          );
+        })}
+      </Stepper>
+      <div>
+        <Button
+          disabled={activeStep === 0}
+          onClick={handleBack}
+          className={classes.button}
+        >
+          Back
+        </Button>
+        <Button
+          variant="contained"
+          color="primary"
+          align="right"
+          onClick={handleNext}
+          className={classes.button}
+        >
+          {activeStep === steps.length - 1 ? "Finish" : "Next"}
+        </Button>
+      </div>
+      <div>
+        {activeStep === steps.length ? (
```

Comment:
this seems to me that you're going a step _after_ the number of steps, right? so activeStep === 3, but activeStep should be the 0-based index of the current step.

Reply:
They are taken to a final page after all the other pages that says you have completed all of the steps would you like to reset and start again (it's like a conclusion page that goes after the steps) -- this seemed like best practice when reading about material stepper but I can remove it

Reply:
It seems weird, but if it's how they do it in React, who am I to judge...